source/distributed/tria_base.cc

```diff
@@ -49,6 +48,9 @@ namespace parallel
     my_subdomain (Utilities::MPI::this_mpi_process (this->mpi_communicator)),
     n_subdomains(Utilities::MPI::n_mpi_processes(mpi_communicator))
   {
+#ifndef DEAL_II_WITH_MPI
+    Assert(false, ExcNotImplemented());
```

Comment:
What about an ```ExcMessage``` explaining that deal.II was configured without MPI support?
As far as I understand where not planning to support the creation of a distributed Triangulation on non-mpi configurations?!